static/js/production.js

```diff
@@ -0,0 +1,198 @@
+        currentAmount.style.color='#FF0066';
+        idName = "full-capacity-time-"+resource.name;
+        const timeLeftToFull = document.getElementById(idName);
+        timeLeftToFull.style.color='#FF0066';
+    }
+};
+
+export const displayDensity = (planet, container) => {
+    var config = {
+        type: 'polarArea',
+        data: getDensityDatasets(planet.resources),
+        options: {
+            responsive: false,
+            legend: {
+                display: false,
+            },
+            scale: {
+                gridLines: {
+                    color: "#1F2125",
+                    lineWidth: 1,
+                },
+                ticks: {
+                    display: false,
+                    beginAtZero: true,
+                    min: 0,
+                    max: 100,
+                    stepSize: 25,
+                    showLabelBackdrop: false,
+                },
+                tooltips: {
+                  enabled: false,
+                },
+                hover: {
+                  display: false,
+                  mode: null,
+                },
+            },
+        }
+    };
+    var ctx = document.getElementById("resource-density");
+
+    ctx.style.height="500px";
+    ctx.style.width="500px";
+
+    var myChart = new Chart(ctx, config);
+    var chartBackground = document.getElementById("chart-background");
+
+    chartBackground.style.height=ctx.style.height;
+    chartBackground.style.width= ctx.style.width;
+    displayDensityPicto(planet.resources);
+};
+
+const getDensityDatasets = resources => {
+    var resourceDataset = { data: [], backgroundColor: [], borderColor: [], borderWidth: 20 };
+    var resData = resourcesData;
+    var added = 0;
+      for (const key in resources) {
+          var resourceTemp = Object.assign({}, resData[resources[key].name]);
+            var dataset = resourceDataset;
+            var density = resources[key].density;
+            var color = resourceTemp.color;
+
+            dataset.data.push(density);
+            dataset.backgroundColor.push(color);
+            dataset.borderColor.push('#1F2125');
+            added ++;
+        }
+        while (added < 6) {
+            var dataset = resourceDataset;
+            var density = 0;
+            var color = "red";
+            dataset.data.push(density);
+            dataset.backgroundColor.push(color);
+            dataset.borderColor.push('#090A0A');
+            added ++;
+        }
+
+    return {
+        datasets: [ resourceDataset],
+    };
+};
+
+const displayDensityPicto = resources => {
+    var added = 0;
+    var chartBackground = document.getElementById("chart-background");
+    var offset = parseInt(chartBackground.style.height);
+    var pictoSize = 84;
+    var imageSize = 64;
+    var borderSize = 10;
+    var resData = resourcesData;
+
+    for (const key in resources) {
+        var pictoBackground = document.createElement('div');
```

Comment:
De manière générale, ne serait-il pas possible de s'occuper des pictos et des datasets dans une même boucle plutôt qu'itérer deux fois sur les mêmes données ? En refactorant le coeur des boucles dans une fonction, on obtient:

```
for (const key in resources) {
    displayPicto(key);
    displayDataset(key);
}
```
Pareil pour le while en dessous